src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -488,6 +509,14 @@ private static void ThrowIfCURLMError(int error)
             }
         }
 
+        private static bool AreEqualNetworkCredentials(NetworkCredential credential1, NetworkCredential credential2)
+        {
+            Debug.Assert(credential1 != null && credential2 != null, "arguments are non-null in network equality check");
+            return credential1.UserName == credential2.UserName &&
+                credential1.Domain == credential2.Domain &&
+                string.Equals(credential1.Password, credential2.Password, StringComparison.Ordinal);
```

Comment:
Understood.  My point was you should be able to just write it as:
```C#
return 
    credential1.UserName == credential2.UserName &&
    credential1.Domain == credential2.Domain &&
    credential1.Password == credential2.Password;
```
My understanding is that there shouldn't be any behavioral difference between:
```C#
string1 == string2
```
and
```C#
string.Equals(string1, string2, StringComparison.Ordinal);
```
@ellismg, am I wrong?